### PR TITLE
Updates to filtering and tier

### PR DESCRIPTION
**Pull request by Alldred**
- Filtering by name still matches both coverpoints and covergroups
- Filtering by tag now only matches against coverpoints

This was because it is intended that matching the name of a covergroup enables all of its children. However since tags accumulate through covergroups, they should default to matching only against coverpoints. 

- set_tier_level now sets a separate active flag within the coverpoint. This allows the tier level to be set independent of any filters applied. Tier can still be used by a filter if required. 

Documentation updated to reflect these changes, as well as some additional description of behaviour

### CODE REVIEW

File: bucket/covergroup.py [hunk 1]
```diff
@@ -78,7 +79,7 @@ def _set_full_path(self):
             cg._set_full_path()
 
     @validate_call
-    def include_by_function(self, matcher: Callable[[CoverBase], bool]):
+    def include_by_function(self, matcher: Callable[[CoverBase], bool], cp_only=False):
```
Comment (Kotarski) [inline bucket/covergroup.py:82]: I don't think it makes sense to add this argument when you can just check in the matcher function with isinstance

**Review by Kotarski — CHANGES_REQUESTED** [review]

File: bucket/covergroup.py (hunk 1 above)
Comment (Alldred) [reply bucket/covergroup.py:82]: Good point. Didn't think of that

File: bucket/covergroup.py [hunk 2]
```diff
@@ -209,16 +216,18 @@ def matcher(cp: CoverBase):
...
+                    for tag in tags:
+                        if tag not in cp.tags:
+                            return False
+                    return True
+                else:
+                    for tag in tags:
+                        if tag in cp.tags:
+                            return True
+                    return False
+            return False
```
Comment (Kotarski) [inline bucket/covergroup.py:230]:
```suggestion
            if not isinstance(cp, Coverpoint):
                return False
            if match_all:
                return all(tag in cp.tags for tag in tags)
            else:
               return any(tag in cp.tags for tag in tags)
```

File: bucket/covergroup.py [hunk 3]
```diff
@@ -192,7 +182,24 @@ def exclude_by_tags(self, tags: TagStrs, match_all: bool = False):
...
+        """
+        self._set_tier_level(tier)
+        return self
+
+    def _set_tier_level(self, tier: int):
+        any_children_active = False
+        for child in self.iter_children():
+            any_children_active |= child._set_tier_level(tier)
+        self.tier_active = any_children_active
+        return self.tier_active
```
Comment (Kotarski) [inline bucket/covergroup.py:202]: I think this is the right approach, but I don't think there's any point having a separate `_set_tier_level` - just put the functionality in `set_tier_level`.

Worth also adding the interface to CoverBase if it's meant to be on both

**Review by Kotarski — CHANGES_REQUESTED** [review]

File: bucket/covergroup.py (hunk 3 above)
Comment (Alldred) [reply bucket/covergroup.py:202]: I split it out into user-facing function and internal function so the internal could be easily changed without needing the user to update their end. 
Also set_tier_level returns self, _set_by_tier returns bool to pass upwards.

Comment (Kotarski) [reply bucket/covergroup.py:202]: 👍 didn't notice the return

**Review by Kotarski — APPROVED** [review]

**Pull request merged**